route/option.go

```diff
@@ -232,6 +236,228 @@ func ServiceGroups(serviceGroups []ServiceGroup) Option {
 	return serviceGroupsFunc(serviceGroups)
 }
 
+// Selector sets the given custom location selector. The location selector lets
+// you define a function which selects the locations that will be inserted next
+// into the solution. If no custom location selector is given, the location with
+// the lowest index will be inserted next.
+func Selector(selector func(PartialPlan) model.Domain) Option {
+	connect()
+	return selectorFunc(selector)
+}
+
+// VehicleUpdater defines an interface that is used to override the vehicle's
+// default value function. The Update function takes a PartialVehicle as an
+// input and returns three values, a VehicleUpdater with potentially updated
+// bookkeeping data, a new solution value and a bool value to indicate wether
+// the vehicle's solution value received an update.
+// See the documentation of Update() for example usage.
+type VehicleUpdater interface {
+	Update(PartialVehicle) (VehicleUpdater, int, bool)
+}
+
+// PlanUpdater defines an interface that is used to override router's default
+// value function. The Update function takes a Plan and a slice of
+// PartialVehicles as an input and returns three values, a PlanUpdater with
+// potentially updated bookkeeping data, a new solution value and a bool value
+// to indicate wether the vehicle's solution value received an update.
+// See the documentation of Update() for example usage.
+type PlanUpdater interface {
+	Update(PartialPlan, []PartialVehicle) (PlanUpdater, int, bool)
+}
+
+/*
+Update sets the collection of functions that are called when transitioning from
```

Comment:
It would be kind of nice to allude to a `store` here, as opposed to `state`.

Reply:
Replaced by store for now.